src/NameService/NameService.cs

```diff
@@ -463,23 +470,38 @@ private static void PostTransfer(UInt160 from, UInt160 to, ByteString tokenId, o
 
         private static bool CheckFragment(string root, bool isRoot)
         {
-            int maxLength = isRoot ? 16 : 62;
+            int maxLength = isRoot ? 16 : 63;
             if (root.Length == 0 || root.Length > maxLength) return false;
             char c = root[0];
             if (isRoot)
             {
-                if (!(c >= 'a' && c <= 'z')) return false;
+                if (!isAlpha(c)) return false;
             }
             else
             {
-                if (!(c >= 'a' && c <= 'z' || c >= '0' && c <= '9')) return false;
+                if (!isAlphaNum(c)) return false;
             }
-            for (int i = 1; i < root.Length; i++)
+            for (int i = 1; i < root.Length - 1; i++)
             {
                 c = root[i];
-                if (!(c >= 'a' && c <= 'z' || c >= '0' && c <= '9')) return false;
+                if (!(isAlphaNum(c) || c == '-')) return false;
             }
-            return true;
+            c = root[root.Length - 1];
+            return isAlphaNum(c);
```

Comment:
The case of `root.Length == 1` needs to be considered.

Reply:
Technically, this case was handled by the existing code, but I've added the explicit `root.Length == 1` check. Fixed.